### PR TITLE
Fix: Add uuid to tube purpose resource

### PR DESCRIPTION
**Pull request by KatyTaylor**
Closes #

#### Changes proposed in this pull request

- it was breaking Limber config generate task as that needed to retrieve the uuid
- copied over from the plate purpose resource

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: app/resources/api/v2/tube_purpose_resource.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ class TubePurposeResource < BaseResource
       # @!attribute [rw]
       # @return [String] The target type.
       attribute :target_type
+
+      # @!attribute [r]
+      # @return [String] gets the UUID of the tube purpose.
```
Comment (sdjmchattie) [inline app/resources/api/v2/tube_purpose_resource.rb:27]:
```suggestion
      # @return [String] The UUID of the tube purpose.
```

**Review by sdjmchattie — CHANGES_REQUESTED** [review]
I think we need to update the tests to make sure this cannot be set during creation or updated.  I think we also need to `default_includes :uuid` near the top of the resource.

**Review by andrewsparkes — APPROVED** [review]

**Comment by KatyTaylor** [comment]
> I think we need to update the tests to make sure this cannot be set during creation or updated. I think we also need to `default_includes :uuid` near the top of the resource.

okidoke

**Review by sdjmchattie — APPROVED** [review]

File: spec/requests/api/v2/tube_purposes_spec.rb [hunk 2]
```diff
@@ -104,6 +107,27 @@
...
+            'attributes' => {
+              'uuid' => updated_uuid
+            }
+          }
+        }
+      end
+
+      it 'returns 400, because uuid is read-only' do
+        api_patch "#{base_endpoint}/#{resource_model.id}", payload
+        expect(response).to have_http_status(400)
```
Comment (andrewsparkes) [inline spec/requests/api/v2/tube_purposes_spec.rb:130]: just fix these 400s as github suggests I think

**Pull request merged**